okta/retry_logic_test.go

```diff
@@ -35,6 +35,42 @@ func Test_429_Will_Automatically_Retry(t *testing.T) {
 	require.Equal(t, 2, info["GET /api/v1/users"], "did not make exactly 2 call to /api/v1/users")
 }
 
+func Test_RateLimitPrevent_Blocks(t *testing.T) {
+	httpmock.Activate()
+	defer httpmock.DeactivateAndReset()
+	configuration, err := NewConfiguration(WithRateLimitPrevent(true))
+	require.NoError(t, err, "Creating a new config should not error")
+	configuration.Debug = true
+	proxyClient := NewAPIClient(configuration)
+	resetTime := time.Now().Add(time.Second * 5)
+	count := 0
+	httpmock.RegisterResponder("GET", "/api/v1/users",
+		func(*http.Request) (*http.Response, error) {
+			if count > 2 {
+				return nil, fmt.Errorf("should not have made more than 2 calls")
+			}
+			count++
+			return MockRateLimitedResponse(count, resetTime), nil
```

Comment:
Didn't use the existing `MockResponse` function as I want the `Date` header to be created after the request was sent so I can check it comes after the reset time